src/main/java/com/commercetools/sync/commons/utils/ChunkUtils.java

```diff
@@ -0,0 +1,77 @@
+package com.commercetools.sync.commons.utils;
+
+import static com.commercetools.sync.commons.utils.CompletableFutureUtils.collectionOfFuturesToFutureOfCollection;
+import static java.util.stream.Collectors.toList;
+
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.client.SphereRequest;
+import io.sphere.sdk.queries.PagedQueryResult;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+public class ChunkUtils {
+
+  /**
+   * Executes the given {@link List} of {@link SphereRequest}s, and collects results in a list.
+   *
+   * @param client {@link SphereClient} responsible for interaction with the target CTP project.
+   * @param requests A list of {@link SphereRequest} implementation to allow {@link SphereClient} to
+   *     execute queries on CTP.
+   * @param <T> the type of the underlying model.
+   * @param <Q> the type of the request model.
+   * @return a list of lists where each list represents the results of passed {@link SphereRequest}.
+   */
+  public static <T, Q extends SphereRequest<T>> CompletableFuture<List<T>> executeChunks(
+      @Nonnull final SphereClient client, @Nonnull final List<Q> requests) {
+
+    final List<CompletableFuture<T>> futures =
+        requests.stream()
+            .map(request -> client.execute(request).toCompletableFuture())
+            .collect(toList());
+
+    return collectionOfFuturesToFutureOfCollection(futures, toList());
+  }
+
+  /**
+   * Flat map the list of lists of {@link PagedQueryResult} to the list of {@link T}.
+   *
+   * @param pagedQueryResults query responses which contains a subset of the matching values.
+   * @param <T> the type of the underlying model.
+   * @return a list of lists where each list represents a chunk of elements with type {@link T}.
+   */
+  public static <T> List<T> flattenPagedQueryResults(
+      @Nonnull final List<PagedQueryResult<T>> pagedQueryResults) {
```

Comment:
This will work only for ``PagedQueryResult`` but not for ``GraphQlBaseResult`` - can you add for the graphQL results too?

Reply:
GraphQlBaseResult is not a generic type, every grapqhl request might have a different flat map implementation.
We could add it when we need it, check:

https://github.com/commercetools/commercetools-sync-java/pull/667/commits/3a0e7a7bab6c96342667a7910bb9d121bfba2206#diff-6139d02a1a0d6b756fbb347aea64dd28e1926a30497415083c9c527cc1becf09R89

Reply:
As i've refactored project-sync to use implementation of ``GraphQlBaseRequest`` which expects always to have implementation of ``GraphQlBaseResult`` i need it.

Reply:
Could you check https://github.com/commercetools/commercetools-sync-java/pull/667/commits/5a0617721df439aa121ade45775fa394c8cb48a0 also considering your changes on project-sync ?